### PR TITLE
Will bump major or minor if user enters sq bracket bump major] or sq …

### PR DESCRIPTION
**Pull request by treeder**
…bracket bump minor]

eg:

"this is my commit [bump minor]" will bump the minor version, rather than patch (default).

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -19,7 +19,24 @@ if [ -z $(grep -m1 -Eo "[0-9]+\.[0-9]+\.[0-9]+" $version_file) ]; then
...
+bump_by="patch"
+last_commit=$(git log -1 --pretty=%B)
+x=$(docker run --rm -i treeder/strings contains "$last_commit" "[bump")
+if [ "$x" == "true" ]; then
+  x=$(docker run --rm -i treeder/strings contains "$last_commit" "[bump minor]")
+  if [ "$x" == "true" ]; then
+    echo "bumping minor version"
+    bump_by="minor"
+  fi
+  x=$(docker run --rm -i treeder/strings contains "$last_commit" "[bump major]")
```
Comment (rdallman) [inline release.sh]: i don't think we need containers to run standard system utilities? seems a bit silly

File: release.sh [hunk 2]
```diff
@@ -19,7 +19,24 @@ if [ -z $(grep -m1 -Eo "[0-9]+\.[0-9]+\.[0-9]+" $version_file) ]; then
...
+  if [ "$x" == "true" ]; then
+    echo "bumping major version"
+    bump_by="major"
+  fi
+fi
+# https://github.com/treeder/dockers/tree/master/bump
+docker run --rm -it -v $PWD:/app -w /app treeder/bump --filename $version_file $bump_by
```
Comment (rdallman) [inline release.sh]: so can we just make this whole script the docker container? and then not shell out to other docker containers and just do the thing? we have this bump script littered across various places now and it's a bit cumbersome, we should just make a docker container that does the thing and run that in our ci.

Comment (treeder) [reply release.sh]: Makes sense.

File: release.sh (hunk 1 above)
Comment (treeder) [reply release.sh]: If by silly you mean awesome. ;) The strings container will contain basically all the Go string function, so you can do a bunch of string ops easily.

**Comment by treeder** [comment]
Going to wrap a bunch of stuff into one container like you said.

**Comment by treeder** [comment]
Alright, check it now.

**Comment by rdallman** [comment]
shwing

**Pull request merged**